### PR TITLE
dsda-doom: init at 0.27.5

### LINKED ISSUE #231121: Package request: dsda-doom
**Project description**

Doom source port. Fork of prboom+ with extra tooling for demo recording and playback, with a focus on speedrunning. 

**Metadata**

* homepage URL:
* source URL: https://github.com/kraflab/dsda-doom
* license: gpl2
* platforms: linux

### PR DESCRIPTION
**Pull request by Gliczy**
## Description of changes
Adds dsda-doom, a popular Doom source port based on PrBoom+ (which is discontinued).
Closes https://github.com/NixOS/nixpkgs/issues/231121

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/games/doom-ports/dsda-doom/default.nix [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+
+stdenv.mkDerivation rec {
+  pname = "dsda-doom";
+  version = "0.27.5";
+
+  src = fetchFromGitHub {
+    owner = "kraflab";
+    repo = "dsda-doom";
+    rev = "v${version}";
+    sha256 = "sha256-+rvRj6RbJ/RaKmlDZdB2oBm/U6SuHNxye8TdpEOZwQw=";
```
Comment (donovanglover) [inline pkgs/games/doom-ports/dsda-doom/default.nix:26]:
```suggestion
    hash = "sha256-+rvRj6RbJ/RaKmlDZdB2oBm/U6SuHNxye8TdpEOZwQw=";
```

File: pkgs/top-level/all-packages.nix [hunk 2]
```diff
@@ -37364,6 +37364,8 @@ with pkgs;
 
   ### GAMES/DOOM-PORTS
 
+  dsda-doom = callPackage ../games/doom-ports/dsda-doom { };
```
Comment (donovanglover) [inline pkgs/top-level/all-packages.nix:37367]: Use [`pkgs/by-name`](https://github.com/NixOS/nixpkgs/tree/master/pkgs/by-name) to avoid editing this file

**Review by donovanglover — COMMENTED** [review]
Welcome to nixpkgs! This builds for me but errors when running:

```
dsda-doom v0.27.5 (https://github.com/kraflab/dsda-doom/)
IdentifyVersion: IWAD not found
```

**Comment by Gliczy** [comment]
I updated the PR according to your suggestions.

The error you are getting is totally normal as you are trying to launch dsda without an IWAD.
IWADs are game files, `DOOM.WAD` for example.

I tested the package myself, but if you want to test it yourself you must either own any classic Doom games or download `Freedoom` which is a free replacement for Doom that can find at https://freedoom.github.io/.

Then, you would have to launch dsda like so `dsda-doom -iwad </path/to/DOOM.WAD>`

**Comment by donovanglover** [comment]
Result of `nixpkgs-review pr 275291` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>dsda-doom</li>
  </ul>
</details>

**Review by donovanglover — APPROVED** [review]
Works for me

**Review by keenanweaver — APPROVED** [review]
Thanks for making this! Also builds and works for me.

File: pkgs/by-name/ds/dsda-doom/package.nix [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+    hash = "sha256-+rvRj6RbJ/RaKmlDZdB2oBm/U6SuHNxye8TdpEOZwQw=";
+  };
+
+  sourceRoot = "${src.name}/prboom2";
+
+  nativeBuildInputs = [
+    cmake
+  ];
+
+   buildInputs = [
```
Comment (pbsds) [inline pkgs/by-name/ds/dsda-doom/package.nix:35]:
```suggestion
  buildInputs = [
```

File: pkgs/by-name/ds/dsda-doom/package.nix [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+  # Fixes impure path to soundfont
+  prePatch = ''
+    substituteInPlace src/m_misc.c --replace \
+      "/usr/share/sounds/sf3/default-GM.sf3" \
+      "${soundfont-fluid}/share/soundfonts/FluidR3_GM2-2.sf2"
+  '';
+
+  meta = with lib; {
+    homepage = "https://github.com/kraflab/dsda-doom";
+    description = "This is a successor of prboom+ with extra tooling for demo recording and playback, with a focus on speedrunning and quality of life";
```
Comment (pbsds) [inline pkgs/by-name/ds/dsda-doom/package.nix:57]:
```suggestion
    description = "A popular Doom source port, successor to PrBoom+";
    longDescription = ''
      A popular Doom source port, successor to PrBoom+ with extra tooling 
      for demo recording and playback, with a focus on speedrunning and 
      quality of life
    '';
```

**Review by pbsds — APPROVED** [review]
Result of `nixpkgs-review pr 275291` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>dsda-doom</li>
  </ul>
</details>

LGTM, save some nits

File: pkgs/by-name/ds/dsda-doom/package.nix (hunk 4 above)
Comment (Gliczy) [reply pkgs/by-name/ds/dsda-doom/package.nix:57]: How about "An advanced Doom source port with a focus on speedrunning, successor of PrBoom+"?

It is shorter than the original description while still conveying it's advanced features and scope.

Comment (pbsds) [reply pkgs/by-name/ds/dsda-doom/package.nix:57]: :+1:

**Pull request merged**